### PR TITLE
ISPN-11950 Create a new module called server-integration module into integrationtests module

### PR DESCRIPTION
**Pull request by gustavolira**
https://issues.redhat.com/browse/ISPN-11950

### CODE REVIEW

File: integrationtests/server-integration/pom.xml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        <ispnserver.dist>${basedir}/target/infinispan-server</ispnserver.dist>
+    </properties>
+
+    <modelVersion>4.0.0</modelVersion>
+    <packaging>pom</packaging>
+
+    <artifactId>infinispan-server-integration</artifactId>
+    <name>Server Integration</name>
+    <description>Server Integration description</description>
+
```
Comment (ryanemerson) [inline integrationtests/server-integration/pom.xml:25]: We should have a `<modules>` section here with all of the child modules.

Comment (gustavolira) [reply integrationtests/server-integration/pom.xml:25]: @ryanemerson are you sure? if look into `integrationtests` or `core` which have child modules it don't have `modules` section, also if you look into `infinispan-parent` I added
```
<module>integrationtests/server-integration</module>
<module>integrationtests/server-integration/wildfly-modules</module>
```

Comment (ryanemerson) [reply integrationtests/server-integration/pom.xml:25]: It's not strictly necessary, but without it it's not possible todo the following:

```
cd integrationtests/server-integration
mvn clean install # All child modules installed
```

In the parent pom.xml, it's then only necessary to add `<module>integrationtests</module>`.

That being said, I see that we don't currently do that for all of the other modules under `integrationtests`, so don't worry about it.

File: integrationtests/server-integration/pom.xml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        <resources.dir>${basedir}/src/test/resources</resources.dir>
+        <ispnserver.project.dir>${basedir}/../../server/runtime</ispnserver.project.dir>
+        <server.build.dist>${ispnserver.project.dir}/target/${infinispan.brand.prefix}-server-${infinispan.brand.version}</server.build.dist>
+        <ispnserver.dist>${basedir}/target/infinispan-server</ispnserver.dist>
+    </properties>
+
+    <modelVersion>4.0.0</modelVersion>
+    <packaging>pom</packaging>
+
+    <artifactId>infinispan-server-integration</artifactId>
```
Comment (ryanemerson) [inline integrationtests/server-integration/pom.xml:22]: `infinispan-server-integration-parent`?

File: pom.xml [hunk 3]
```diff
@@ -145,7 +145,8 @@
       <module>integrationtests/security-manager-it</module>
       <module>integrationtests/cdi-weld-se-it</module>
       <module>integrationtests/spring-boot-it</module>
-      <module>integrationtests/wildfly-modules</module>
+      <module>integrationtests/server-integration</module>
+      <module>integrationtests/server-integration/wildfly-modules</module>
```
Comment (gustavolira) [inline pom.xml:149]: @ryanemerson

File: integrationtests/server-integration/pom.xml (hunk 1 above)
Comment (gustavolira) [reply integrationtests/server-integration/pom.xml:25]: @ryanemerson I just added the `modules` section as you suggested

**Comment by gustavolira** [comment]
Hey @tristantarrant can you review this PR?

**Comment by ryanemerson** [comment]
@gustavolira The build is failing as the versions of the `pom.xml` needs to be updated to `12.0.0-SNAPSHOT`.

**Comment by gustavolira** [comment]
> @gustavolira The build is failing as the versions of the added `pom.xml` files need to be updated to `12.0.0-SNAPSHOT`.

Thanks @ryanemerson, I just fixed

**Comment by ryanemerson** [comment]
@gustavolira The execution time of the run is way too quick, I don't think all of the tests are being executed like you expect:

```
14:31:46  09:31:46.193 [INFO] Infinispan WildFly Feature Pack .................... SUCCESS [  4.067 s]
14:31:46  09:31:46.193 [INFO] Infinispan Integration Tests Parent ................ SUCCESS [  0.071 s]
14:31:46  09:31:46.193 [INFO] Integration tests - Endpoints interop .............. SUCCESS [ 17.334 s]
14:31:46  09:31:46.193 [INFO] Integration tests - JBoss Marshalling interop ...... SUCCESS [  9.310 s]
14:31:46  09:31:46.194 [INFO] Integration tests - CDI/JCache interactions ........ SUCCESS [  6.585 s]
14:31:46  09:31:46.194 [INFO] Integration tests - Security Tests ................. SUCCESS [04:36 min]
14:31:46  09:31:46.194 [INFO] Integration tests - SecurityManager tests .......... SUCCESS [ 35.785 s]
14:31:46  09:31:46.194 [INFO] Integration tests - Weld SE CDI tests .............. SUCCESS [  3.599 s]
14:31:46  09:31:46.194 [INFO] Integration tests - Spring Boot .................... SUCCESS [  9.408 s]
14:31:46  09:31:46.194 [INFO] Server Integration ................................. SUCCESS [ 12.072 s]
14:31:46  09:31:46.195 [INFO] Integration tests - WildFly Module Integration Tests SUCCESS [  4.970 s]

```

**Comment by gustavolira** [comment]
> @gustavolira The execution time of the run is way too quick, I don't think all of the tests are being executed like you expect:
> 
> ```
> 14:31:46  09:31:46.193 [INFO] Infinispan WildFly Feature Pack .................... SUCCESS [  4.067 s]
> 14:31:46  09:31:46.193 [INFO] Infinispan Integration Tests Parent ................ SUCCESS [  0.071 s]
> 14:31:46  09:31:46.193 [INFO] Integration tests - Endpoints interop .............. SUCCESS [ 17.334 s]
> 14:31:46  09:31:46.193 [INFO] Integration tests - JBoss Marshalling interop ...... SUCCESS [  9.310 s]
> 14:31:46  09:31:46.194 [INFO] Integration tests - CDI/JCache interactions ........ SUCCESS [  6.585 s]
> 14:31:46  09:31:46.194 [INFO] Integration tests - Security Tests ................. SUCCESS [04:36 min]
> 14:31:46  09:31:46.194 [INFO] Integration tests - SecurityManager tests .......... SUCCESS [ 35.785 s]
> 14:31:46  09:31:46.194 [INFO] Integration tests - Weld SE CDI tests .............. SUCCESS [  3.599 s]
> 14:31:46  09:31:46.194 [INFO] Integration tests - Spring Boot .................... SUCCESS [  9.408 s]
> 14:31:46  09:31:46.194 [INFO] Server Integration ................................. SUCCESS [ 12.072 s]
> 14:31:46  09:31:46.195 [INFO] Integration tests - WildFly Module Integration Tests SUCCESS [  4.970 s]
> ```

@ryanemerson good catch, I just update, thanks!

**Comment by ryanemerson** [comment]
Thanks @gustavolira

**Pull request merged**